library/core/src/cell.rs

```diff
@@ -1357,6 +1358,7 @@ impl<'b, T: ?Sized> Ref<'b, T> {
     /// assert_eq!(*b2, 5)
     /// ```
     #[stable(feature = "cell_map", since = "1.8.0")]
+    #[must_use = "should not be called for side effects"]
```

Comment:
While `Iterator::map` doesn't do anything unless you use the resulting iterator, functions like `Cell::map` do actually call their function, so it *is* legitimate (even if undesirable) to use them for their side effects. I think you should omit these and similar functions that call a closure, for which it seems plausible to use a closure with side effects.

Reply:
Removed.